### DISCUSSION TITLE
implement get unique constraints for MSSQL dialect

### DISCUSSION THREAD

**Discussion by bfmcneill** [sqlalchemy/sqlalchemy#6555] [Usage Questions] [answered]
I tried to inspect the unique constraints for MSSQL dialect but a `NotImplementedError` was raised.

**Comment by zzzeek** [comment]
cc @gordthompson  does SQL Server have a unique constraint concept separate from unique indexes?

**Comment by zzzeek** [comment, marked as answer]
per https://www.mssqltips.com/sqlservertip/4270/difference-between-sql-server-unique-indexes-and-unique-constraints/ it looks like these are identitical and that a unique index is implicit.  you should be able to use get_indexes() and look for those that have unique=True.

**Reply by CaselIT** [reply]
it does end up in the indexes of the table. Example `print(table.indexes)`

```
{Index('UQ__a__D9DE21E17ED33B44', Column('data', VARCHAR(length=100, collation='SQL_Latin1_General_CP1_CI_AS'), table=<a>, server_default=DefaultClause(<sqlalchemy.sql.elements.TextClause object at 0x000002467602BA00>, for_update=False)), unique=True)}
```

**Comment by bfmcneill** [comment]
yes and thank you.
